dolphinscheduler-api-test/dolphinscheduler-api-test-case/src/test/java/org/apache/dolphinscheduler/api.test/pages/project/WorkFlowDefinitionPage.java

```diff
@@ -0,0 +1,205 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.dolphinscheduler.api.test.pages.project;
+
+
+import org.apache.dolphinscheduler.api.test.cases.TenantAPITest;
+import org.apache.dolphinscheduler.api.test.core.Constants;
+import org.apache.dolphinscheduler.api.test.entity.TaskParamsMap;
+import org.apache.dolphinscheduler.api.test.entity.TaskDefinitionRequestData;
+import org.apache.dolphinscheduler.api.test.entity.WorkFlowRunRequestData;
+import org.apache.dolphinscheduler.api.test.entity.HttpResponse;
+import org.apache.dolphinscheduler.api.test.entity.WorkFlowCreateRequestData;
+import org.apache.dolphinscheduler.api.test.entity.TaskRelationRequestData;
+import org.apache.dolphinscheduler.api.test.entity.WorkFlowResponseData;
+import org.apache.dolphinscheduler.api.test.entity.WorkFlowResponseTotalList;
+import org.apache.dolphinscheduler.api.test.utils.JSONUtils;
+import org.apache.dolphinscheduler.api.test.utils.RequestClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+
+
+public final class WorkFlowDefinitionPage {
+    private static String genNumId = null;
+    private static String workFlowCode = null;
+    private static final Logger logger = LoggerFactory.getLogger(WorkFlowDefinitionPage.class);
+
+    public void getGenNumId(String sessionId, String projectName) {
+
+        Map<String, Object> params = new HashMap<>();
+        Map<String, String> headers = new HashMap<>();
+        params.put("genNum", 1);
+        headers.put(Constants.SESSION_ID_KEY, sessionId);
+        ProjectPage project = new ProjectPage();
+        String projectCode = project.getProjectCode(sessionId, projectName);
+        RequestClient requestClient = new RequestClient();
+        HttpResponse res = requestClient.get("/projects/"+projectCode+"/task-definition/gen-task-codes", headers, params);
+        ArrayList list = (ArrayList) res.body().data();
+        genNumId = list.get(0).toString();
+
+    }
+
+
+    public HttpResponse createWorkflow(String sessionId, String projectName, String workFlowName) {
+        Map<String, String> headers = new HashMap<>();
+        WorkFlowCreateRequestData workFlowCreateRequestData = new WorkFlowCreateRequestData();
+        TaskDefinitionRequestData taskDefinitionRequestData = new TaskDefinitionRequestData();
+        TaskRelationRequestData taskRelationRequestData = new TaskRelationRequestData();
+
+        TaskParamsMap taskParams = new TaskParamsMap();
+        ArrayList<Object> localParams = new ArrayList<>();
+        ArrayList<Object> resourceList = new ArrayList<>();
+        taskParams.localParams(localParams);
+        taskParams.resourceList(resourceList);
+        taskParams.rawScript("echo 123");
```

Comment:
We should turn this into an input to the method instead of hardcode it.

Reply:
Thanks for the reminder, it has been fixed